File: app/src/main/java/com/rolandvitezhu/todocloud/fragment/ReminderTimePickerDialogFragment.java
package com.rolandvitezhu.todocloud.fragment;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Bundle;
import android.support.v7.app.AppCompatDialogFragment;
import android.widget.TimePicker;

import com.rolandvitezhu.todocloud.R;
import com.rolandvitezhu.todocloud.app.Constant;

import org.threeten.bp.LocalDateTime;

public class ReminderTimePickerDialogFragment extends AppCompatDialogFragment implements TimePickerDialog.OnTimeSetListener {

  private int hour;
  private int minute;
  private LocalDateTime date;

  @Override
  public Dialog onCreateDialog(Bundle savedInstanceState) {
    if (getArguments() != null) {
      date = (LocalDateTime) getArguments().get(Constant.REMINDER_DATE_TIME);

      if (date != null) {
        hour = date.getHour();
        minute = date.getMinute();
      }
    }
    return new TimePickerDialog(
        getActivity(), R.style.MyPickerDialogTheme, this, hour, minute, true
    );
  }

  @Override
  public void onTimeSet(TimePicker view, int hour, int minute) {
    date = LocalDateTime.of(date.getYear(), date.getMonth(), date.getDayOfMonth(), hour, minute);

    if (getTargetFragment() instanceof CreateTodoFragment)
      ((CreateTodoFragment)getTargetFragment()).onSelectReminderDateTime(date);
    else if (getTargetFragment() instanceof ModifyTodoFragment)
      ((ModifyTodoFragment)getTargetFragment()).onSelectReminderDateTime(date);

    dismiss();
  }

}

File: app/src/main/java/com/rolandvitezhu/todocloud/ui/activity/main/dialogfragment/ReminderTimePickerDialogFragment.kt
package com.rolandvitezhu.todocloud.ui.activity.main.dialogfragment

import android.app.Dialog
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import android.widget.TimePicker
import androidx.appcompat.app.AppCompatDialogFragment
import androidx.lifecycle.ViewModelProvider
import com.rolandvitezhu.todocloud.R
import com.rolandvitezhu.todocloud.ui.activity.main.viewmodel.TodosViewModel
import org.threeten.bp.LocalDateTime
import org.threeten.bp.Month

class ReminderTimePickerDialogFragment : AppCompatDialogFragment(), OnTimeSetListener {

    private val todosViewModel by lazy {
        ViewModelProvider(requireActivity()).get(TodosViewModel::class.java)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        return TimePickerDialog(
                requireActivity(),
                R.style.MyPickerDialogTheme,
                this,
                todosViewModel.ldtReminderDateTime?.hour ?: 0,
                todosViewModel.ldtReminderDateTime?.minute ?: 0,
                true)
    }

    override fun onTimeSet(view: TimePicker, hour: Int, minute: Int) {
        todosViewModel.ldtReminderDateTime =
                LocalDateTime.of(
                        todosViewModel.ldtReminderDateTime?.year ?: 0,
                        todosViewModel.ldtReminderDateTime?.month ?: Month.of(1),
                        todosViewModel.ldtReminderDateTime?.dayOfMonth ?: 0,
                        hour,
                        minute)

        todosViewModel.setTodoReminderDateTime(todosViewModel.ldtReminderDateTime)

        dismiss()
    }
}